site/templates/common/cta_buttons.jet

```diff
@@ -12,13 +12,21 @@
 
   {{showPricingButtons := isFilm || isSeason || isBundle}}
   {{showPlayButton := isFilm || isSeason || isEpisode}}
-  {{showCanBeWatchedButton := isFilm}}
   {{showTrailerButton := trailerURL != ""}}
   {{showWishlistButton := isFilm || isSeason || isEpisode}}
   {{showShareButton := isFilm || isSeason || isEpisode || isBundle}}
   {{autoplay := !isCarousel}}
 
-  {{showCtaButtons := showPricingButtons || showPlayButton || showCanBeWatchedButton || showTrailerButton || showWishlistButton || showShareButton}}
+  
+  {{canBeWatchedButtonLink := .CustomFields.GetString("can_be_watched_button_link", "")}}
+  {{canBeWatchedButtonText := .CustomFields.GetString("can_be_watched_button_text", "can_be_watched_button_text")}}
+  {{showCanBeWatchedButton := len(canBeWatchedButtonLink) > 0}}
```

Comment:
I believe that `isFilm` was a safety against clients filling out custom fields on item-types that can't support the can-be-watched-button... e.g. custom fields aren't item-type specific so a client could mistakenly fill this out for a `page` which will make the can-be-watched-button render despite a page not having watchable content.

Could do: 
```
{{showCanBeWatchedButton := isFilm && len(canBeWatchedButtonLink) > 0}}
```

Reply:
Just another reason why custom fields were a mistake.

Reply:
I figured that might have been the rationale, but it seemed harmless and possibly made the hack buttons more useful?

I guess it's a problem if there are items that might not have a CustomFields map. Plans don't, maybe other generic item types don't and that could crash the site build?

If that's the case then, would `{{hasCustomFields := isset(.CustomFields)}}` be a sensible guard ?

Reply:
I don't think `isset(.CustomFields)` is a great check because you would get false-positives from custom fields unrelated to this "feature".

Yeah I guess it is harmless as long as the client is paying attention to what they copy-paste into those custom fields... if they aren't then I foresee support requests!